rest-service/manager_rest/rest/responses.py

```diff
@@ -153,17 +156,23 @@ def __init__(self, **kwargs):
 
 
 @swagger.model
-class Label(object):
-
-    resource_fields = {
+@dataclass(unsafe_hash=True)
+class Label:
+    """A blueprint or deployment label."""
+    key: str
+    value: str
+    created_at: Optional[str | datetime] = field(default=None, compare=False)
+    created_by: Optional[str] = field(default=None, compare=False)
+
+    resource_fields: ClassVar[Mapping] = {
         'key': fields.String,
         'value': fields.String,
         'created_at': fields.String,
         'created_by': fields.String,
     }
 
-    def __init__(self, **kwargs):
-        self.key = kwargs.get('key')
-        self.value = kwargs.get('value')
-        self.created_at = kwargs.get('created_at')
-        self.created_by = kwargs.get('created_by')
+    def to_dict(self) -> dict:
+        return {'key': self.key,
+                'value': self.value,
+                'created_at': self.created_at,
+                'created_by': self.created_by}
```

Comment:
`dataclasses.asdict` perhaps

Reply:
Note: After discussion this should perhaps use asdict within the method (so that it need not be imported elsewhere, but so that we get the behaviour even if we update the dataclass later).

Reply:
Updated